snewpdag/values/Hist1D.py

```diff
@@ -39,22 +49,32 @@ def is_compatible(self, other):
     else:
       return False
 
+  def bin(self, x):
+    try:
+      return self.nbins * (x - self.xlow) / self.xwidth
+    except:
+      logging.info('Hist1D.bin: index calc error {}'.format(sys.exc_info()))
+      return None
+
   def fill(self, x, weight=1.0):
     """
     fill histogram.  x can be a scalar or an array of fill values.
     """
     v = np.array(x)
-    try:
-      ix = self.nbins * (v - self.xlow) / self.xwidth
-    except:
-      logging.info('Hist1D: index calculation error {}'.format(sys.exc_info()))
-      return
-    h, bin_edges = np.histogram(ix, bins=self.nbins, range=(0, self.nbins))
-    self.bins += h
-    self.underflow += weight * np.sum(ix < 0)
-    self.overflow += weight * np.sum(ix >= self.nbins)
-    self.sum += np.sum(v)
-    self.sum2 += np.sum(v*v)
+    #try:
+    #  ix = self.nbins * (v - self.xlow) / self.xwidth
+    #except:
+    #  logging.info('Hist1D: index calculation error {}'.format(sys.exc_info()))
+    #  return
+    #h, bin_edges = np.histogram(ix, bins=self.nbins, range=(0, self.nbins))
+    h, edges = np.histogram(x, bins=self.nbins, range=(self.xlow, self.xhigh))
```

Comment:
Not too relevant but, why Hist1D is in values? In which sense is a value? I wouldn't expect to find it there

Reply:
"values" is a collection of objects which can be stored in the payload. (Good question, though - finding the right word helps everyone, but it takes some effort!)